In [1]:
import pandas as pd

In [2]:
import numpy as np

# doc_list

In [3]:
doc_list = pd.read_csv("doctors_list_t8.csv")

In [4]:
pd.isnull(doc_list).sum()

Id             0
initial        0
href           0
name           0
specialties    0
address        0
zip            0
dtype: int64

In [4]:
doc_list.head()

Id initial                         href               name  \
0  3886560       a  /doctors/Dr_Enenge_ABodjedi    Enenge Abodjedi   
1  3886561       a     /doctors/Dr_Brady_AHearn    Brady J A'Hearn   
2  3886562       a    /doctors/Dr_Yousif_Arahim     Yousif A-Rahim   
3  3886563       a    /doctors/Dr_Noor_Al_Khori      Noor Al Khori   
4  3886564       a  /doctors/Dr_Salam_Al-Hafidh  Salam N Al-Hafidh   

                                        specialties                   address  \
0                                  ['Psychiatrist']        Stamford, CT 06902   
1                                  ['Pediatrician']       Iowa City, IA 52242   
2  ['Gastroenterologist (digestive)', ' Internist']            Aiea, HI 96701   
3                                       ['Student']    Grand Rapids, MI 49503   
4                                  ['Pediatrician']  Richland Hills, TX 76180   

     zip  
0  06902  
1  52242  
2  96701  
3  49503  
4  76180

In [5]:
doc_list.tail()

Id initial                        href            name  \
1294498  5181716       i   /dentists/Dr_Micheal_Izzo  Michael A Izzo   
1294499  5181717       i     /doctors/Dr_Philip_Izzo   Philip M Izzo   
1294500  5181718       i    /dentists/Dr_Steven_Izzo   Steven R Izzo   
1294501  5181719       i  /dentists/Dr_Steven_Izzo_2     Steven Izzo   
1294502  5181720       i    /doctors/Dr_Timothy_Izzo  Timothy J Izzo   

                                             specialties  \
1294498                                      ['Dentist']   
1294499                                 ['Psychiatrist']   
1294500                                      ['Dentist']   
1294501                                      ['Dentist']   
1294502  ['Family Practitioner', ' Primary Care Doctor']   

                        address    zip  
1294498  Lincoln Park, MI 48146  48146  
1294499         Media, PA 19063  19063  
1294500     Ridgewood, NY 11385  11385  
1294501      Brooklyn, NY 11228  11228  
1294502   Grand Ledge, MI 48837  48837

In [6]:
np.shape(doc_list) # see the number of rows and columns

(1294503, 7)

- We have 1294503 doctors in the data set

In [7]:
r = lambda x: x.lstrip("/") 

In [8]:
doc_list.href = doc_list.href.apply(r) # split the href into title and the prof_name

In [9]:
doc_list['title'], doc_list['prof_name'] = doc_list['href'].str.split('/',1).str

In [10]:
doc_list = doc_list.drop('href',1)

In [11]:
doc_list.tail() 

Id initial            name  \
1294498  5181716       i  Michael A Izzo   
1294499  5181717       i   Philip M Izzo   
1294500  5181718       i   Steven R Izzo   
1294501  5181719       i     Steven Izzo   
1294502  5181720       i  Timothy J Izzo   

                                             specialties  \
1294498                                      ['Dentist']   
1294499                                 ['Psychiatrist']   
1294500                                      ['Dentist']   
1294501                                      ['Dentist']   
1294502  ['Family Practitioner', ' Primary Care Doctor']   

                        address    zip     title         prof_name  
1294498  Lincoln Park, MI 48146  48146  dentists   Dr_Micheal_Izzo  
1294499         Media, PA 19063  19063   doctors    Dr_Philip_Izzo  
1294500     Ridgewood, NY 11385  11385  dentists    Dr_Steven_Izzo  
1294501      Brooklyn, NY 11228  11228  dentists  Dr_Steven_Izzo_2  
1294502   Grand Ledge, MI 48837  48837   doctors   Dr_Timothy_Izzo

In [17]:
doc_list.title.value_counts()

doctors     1095990
dentists     198513
Name: title, dtype: int64

There are 1095990 doctors and 198513 dentists

In [12]:
import re

In [24]:
rs = lambda x: re.sub('[^a-zA-Z\']+', ' ', x)

In [25]:
doc_list.specialties = doc_list.specialties.apply(rs) # clean the specialties columns

In [26]:
doc_list.head()

Id initial               name  \
0  3886560       a    Enenge Abodjedi   
1  3886561       a    Brady J A'Hearn   
2  3886562       a     Yousif A-Rahim   
3  3886563       a      Noor Al Khori   
4  3886564       a  Salam N Al-Hafidh   

                                specialties                   address    zip  \
0                            'Psychiatrist'        Stamford, CT 06902  06902   
1                            'Pediatrician'       Iowa City, IA 52242  52242   
2  'Gastroenterologistdigestive''Internist'            Aiea, HI 96701  96701   
3                                 'Student'    Grand Rapids, MI 49503  49503   
4                            'Pediatrician'  Richland Hills, TX 76180  76180   

     title           prof_name  
0  doctors  Dr_Enenge_ABodjedi  
1  doctors     Dr_Brady_AHearn  
2  doctors    Dr_Yousif_Arahim  
3  doctors    Dr_Noor_Al_Khori  
4  doctors  Dr_Salam_Al-Hafidh

In [21]:
np.shape(doc_list)

(1294503, 8)

In [31]:
words = []
for i in doc_list.specialties:
    words += [i]

In [35]:
words_clean = re.sub('[^a-zA-Z]+', '/', str(words))

In [37]:
#count the specialties
specialist = {}
wordsplit = words_clean.split('/')
for word in wordsplit:
    if word in specialist :
        specialist[word]=specialist[word]+1
    else:
        specialist[word] = 1
     

In [46]:
sorted(specialist.items(), key=lambda x: x[1],reverse=True)

[('Internist', 222313),
 ('Dentist', 198510),
 ('PrimaryCareDoctor', 139732),
 ('FamilyPractitioner', 131692),
 ('SurgicalSpecialist', 100403),
 ('Pediatrician', 92566),
 ('Psychologist', 87175),
 ('Psychiatrist', 57374),
 ('Student', 56887),
 ('ObstetricianGynecologistOBGYN', 52807),
 ('Anesthesiologistpaincontrol', 52793),
 ('Radiologist', 48060),
 ('EmergencyPhysician', 46557),
 ('OrthopaedicSurgeon', 31876),
 ('Cardiologistheart', 31593),
 ('Pathologist', 23725),
 ('Ophthalmologisteyes', 23133),
 ('OncologySpecialistcancer', 22076),
 ('Neurologistbrain', 20461),
 ('nervoussystem', 20460),
 ('Gastroenterologistdigestive', 16900),
 ('PediatricSpecialist', 15693),
 ('GeneralPractitioner', 15451),
 ('Dermatologistskin', 14867),
 ('Pulmonologistlungs', 14723),
 ('Otolaryngologistear', 13356),
 ('nose', 13356),
 ('throat', 13356),
 ('Urologist', 13183),
 ('CriticalCareSpecialist', 11958),
 ('GeriatricSpecialistelderlycare', 11838),
 ('Physiatristphysical', 11498),
 ('rehabilitation', 114

the TOP 10 	specialties are:
 - ('Internist', 222313),
 - ('Dentist', 198510),
 - ('PrimaryCareDoctor', 139732),
 - ('FamilyPractitioner', 131692),
 - ('SurgicalSpecialist', 100403),
 - ('Pediatrician', 92566),
 - ('Psychologist', 87175),
 - ('Psychiatrist', 57374),
 - ('Student', 56887),
 - ('ObstetricianGynecologistOBGYN', 52807)

In [47]:
specialist_table = pd.DataFrame.from_dict(specialist, orient='index')

# doc_panel

In [4]:
doc_panel = pd.read_csv("doctors_panel_t8.csv")

In [5]:
doc_panel.head()

Id                                href  years_of_exp  overall_rating  \
0  9206066         /dentists/Dr_Aamir_M_Sheikh          10.0             0.0   
1  9206067           /dentists/Dr_Aamir_Shah_2           NaN             4.5   
2  9206068            /dentists/Dr_Aalie_Sagar           NaN             3.5   
3  9206069           /dentists/Dr_Aan_Litvinov           NaN             1.0   
4  9206070  /dentists/Dr_Aarika_Anderson_Elter           NaN             5.0   

   n_ratings  n_reviews avg_wating  Ease_of_Appointment  Promptness  \
0          0          0        NaN                  NaN         NaN   
1          3          0        NaN                  NaN         NaN   
2          3          0        NaN                  NaN         NaN   
3          1          0        NaN                  NaN         NaN   
4          1          1        NaN                  3.5         NaN   

   Courteous_Staff         ...           Caring_Manner  Pain_Minimized  \
0              NaN         ...                     NaN             NaN   
1              NaN         ...                     NaN             NaN   
2              NaN         ...                     NaN             NaN   
3              NaN         ...                     NaN             NaN   
4              NaN         ...                     5.0             5.0   

   Satisfaction  n_5  n_4  n_3  n_2  n_1  \
0           NaN  NaN  NaN  NaN  NaN  NaN   
1           NaN  NaN  NaN  NaN  NaN  NaN   
2           NaN  NaN  NaN  NaN  NaN  NaN   
3           NaN  NaN  NaN  NaN  NaN  NaN   
4           NaN  1.0  0.0  0.0  0.0  0.0   

                                     address           crawl_date  
0      9105 All Saints Rd, Laurel, MD, 20723  2016-09-06 20:53:49  
1        7872 Walker St, La Palma, CA, 90623  2016-09-06 20:53:51  
2  10842 Potranco Rd, San Antonio, TX, 78251  2016-09-06 20:53:54  
3       1740 Culver Rd, Rochester, NY, 14609  2016-09-06 20:53:55  
4          1708 E 44th St, Tacoma, WA, 98404  2016-09-06 20:54:01  

[5 rows x 27 columns]

In [11]:
doc_panel.tail()

Id                        href  years_of_exp  overall_rating  \
1293677  10499745       /doctors/Dr_Ramy_Doss          25.0             4.5   
1293678  10499746    /doctors/Dr_Ramzi_Khazen          11.0             0.0   
1293679  10499747   /doctors/Dr_Randall_Edson          37.0             3.0   
1293680  10499748  /doctors/Dr_Randall_Farrar          26.0             4.5   
1293681  10499749  /doctors/Dr_Randall_Finken          49.0             5.0   

         n_ratings  n_reviews  avg_wating  Ease_of_Appointment  Promptness  \
1293677         21         11  33 minutes                  5.0         4.0   
1293678          0          0         NaN                  NaN         NaN   
1293679          2          1  10 minutes                  5.0         3.5   
1293680         18          1  17 minutes                  4.0         4.5   
1293681          2          0         NaN                  NaN         NaN   

         Courteous_Staff         ...           Caring_Manner  Pain_Minimized  \
1293677              4.5         ...                     NaN             NaN   
1293678              NaN         ...                     NaN             NaN   
1293679              5.0         ...                     NaN             NaN   
1293680              4.5         ...                     NaN             NaN   
1293681              NaN         ...                     NaN             NaN   

         Satisfaction   n_5  n_4  n_3  n_2  n_1  \
1293677           NaN  16.0  2.0  1.0  2.0  0.0   
1293678           NaN   NaN  NaN  NaN  NaN  NaN   
1293679           NaN   1.0  0.0  0.0  0.0  1.0   
1293680           NaN  14.0  1.0  0.0  0.0  3.0   
1293681           NaN   NaN  NaN  NaN  NaN  NaN   

                                        address           crawl_date  
1293677  10293 N 92nd St, Scottsdale, AZ, 85258  2016-09-16 20:03:43  
1293678   60 Hospital Rd, Leominster, MA, 01453  2016-09-16 20:03:45  
1293679     200 1st St Sw, Rochester, MN, 55905  2016-09-16 20:03:45  
1293680    1155 Jadwin Ave, Richland, WA, 99352  2016-09-16 20:03:47  
1293681      3909 Woodley Rd, Toledo, OH, 43606  2016-09-16 20:03:48  

[5 rows x 27 columns]

In [7]:
rew_panel = pd.read_csv("reviews_panel_t8.csv")

In [8]:
rew_panel.head()

Id  review_id                                href  overall_rating  \
0  12949966   23507540  /dentists/Dr_Aarika_Anderson_Elter               5   
1  12949967   27184431            /dentists/Dr_Aamir_Wahab               5   
2  12949968   26307282           /dentists/Dr_Aanal_Parikh               1   
3  12949969   28904504          /dentists/Dr_Aaron_Aguilar               5   
4  12949970   28380953         /dentists/Dr_Aaron_D_Larsen               4   

   Ease_of_Appointment  Promptness  Courteous_Staff  Accurate_Diagnosis  \
0                  4.0         NaN              NaN                 NaN   
1                  5.0         NaN              NaN                 NaN   
2                  1.0         NaN              NaN                 NaN   
3                  5.0         NaN              NaN                 NaN   
4                  3.0         NaN              NaN                 NaN   

   Bedside_Manner  Spends_Time_with_Me         ...           \
0             NaN                  NaN         ...            
1             NaN                  NaN         ...            
2             NaN                  NaN         ...            
3             NaN                  NaN         ...            
4             NaN                  NaN         ...            

   Advanced_Technology  Caring_Manner  Pain_Minimized  Satisfaction  \
0                  NaN            5.0             5.0           NaN   
1                  5.0            5.0             5.0           5.0   
2                  1.0            1.0             1.0           1.0   
3                  5.0            5.0             5.0           5.0   
4                  4.0            4.0             4.0           5.0   

         date      reviewer                                 title  \
0  2012-11-16           NaN                        Great results    
1  2015-03-15       Brad s.                               implant   
2  2014-10-12  Bill johnson  Warning ..would not see this dentist   
3  2015-06-11           NaN                                   NaN   
4  2015-03-31           NaN             Dr larsen great others no   

                                             content helpful_vote  \
0  Dr. Anderson explained in detail my options fo...          NaN   
1  Had an implant done and it was painless. I cou...          NaN   
2  Do not go to this dentist office they scam peo...          NaN   
3  Very thoughtful Dr.  Communicates/bedside mann...          NaN   
4  Would continue to see Dr.  Larsen however his ...          NaN   

            crawl_date  
0  2016-09-06 20:54:01  
1  2016-09-06 20:54:10  
2  2016-09-06 20:54:13  
3  2016-09-06 20:54:14  
4  2016-09-06 20:54:18  

[5 rows x 23 columns]

In [12]:
rew_panel.tail()

Id  review_id                        href  overall_rating  \
1796199  14819364   19546944       /doctors/Dr_Ramy_Doss               5   
1796200  14819365   18196930       /doctors/Dr_Ramy_Doss               5   
1796201  14819366   18233638       /doctors/Dr_Ramy_Doss               5   
1796202  14819367   19725098   /doctors/Dr_Randall_Edson               5   
1796203  14819368   31104374  /doctors/Dr_Randall_Farrar               5   

         Ease_of_Appointment  Promptness  Courteous_Staff  Accurate_Diagnosis  \
1796199                  5.0         4.0              5.0                 5.0   
1796200                  5.0         4.0              5.0                 5.0   
1796201                  5.0         4.0              5.0                 5.0   
1796202                  5.0         4.0              5.0                 5.0   
1796203                  5.0         5.0              5.0                 5.0   

         Bedside_Manner  Spends_Time_with_Me         ...           \
1796199             5.0                  5.0         ...            
1796200             5.0                  5.0         ...            
1796201             5.0                  5.0         ...            
1796202             5.0                  5.0         ...            
1796203             5.0                  NaN         ...            

         Advanced_Technology  Caring_Manner  Pain_Minimized  Satisfaction  \
1796199                  NaN            NaN             NaN           NaN   
1796200                  NaN            NaN             NaN           NaN   
1796201                  NaN            NaN             NaN           NaN   
1796202                  NaN            NaN             NaN           NaN   
1796203                  NaN            NaN             NaN           NaN   

               date        reviewer                     title  \
1796199  2011-07-13             NaN                       NaN   
1796200  2008-06-23         Cheri R  Wonderful, caring Doctor   
1796201  2008-04-30  THERESA DALTON                       NaN   
1796202  2011-11-21             NaN                       NaN   
1796203  2015-12-19             NaN                       NaN   

                                                   content helpful_vote  \
1796199  Love  Dr. Army  Doss.  He saved my life- a hea...          NaN   
1796200  I met Dr. Doss during a very stressful time an...          NaN   
1796201  IF IT WEREN'T FOR DR. DOSS' ACCURATE DIAGNOSIS...          NaN   
1796202  dr. edson is as good as it gets for being pers...          1.0   
1796203                           Great Doc! Really cares.          NaN   

                  crawl_date  
1796199  2016-09-16 20:03:44  
1796200  2016-09-16 20:03:44  
1796201  2016-09-16 20:03:44  
1796202  2016-09-16 20:03:45  
1796203  2016-09-16 20:03:47  

[5 rows x 23 columns]